### ISSUE TITLE
iSCSI external-provisioner returns empty string in Portals even if there isn't additional portals

### ISSUE THREAD

**Issue by mtanino** [kubernetes-incubator/external-storage#431]
At the API Spec, iSCSI plugin expects that Portals is empty if there isn't no additional portals.
```
// iSCSI target portal List. The portal is either an IP or ip_addr:port if the port
// is other than default (typically TCP ports 860 and 3260).
// +optional
Portals []string `json:"portals,omitempty" protobuf:"bytes,7,opt,name=portals"`
```

However, iscsi/targetd external-provisioner set empty string if there isn't additional portals.
As a result, iscsi plugin misunderstands that this empty string is actual second portal, then trying to connect ":3260" iscsi portal as shown in iscsi plugin error below.

- Provision() method in iscsi/targetd external-provisioner
```
    PersistentVolumeSource: v1.PersistentVolumeSource{
          ISCSI: &v1.ISCSIVolumeSource{
               TargetPortal:      options.Parameters["targetPortal"],
               Portals:           strings.Split(options.Parameters["portals"], "'"),
               IQN:               options.Parameters["iqn"],
               ISCSIInterface:    options.Parameters["iscsiInterface"],
               Lun:               lun,
```

- iscsi plugin error
```
E1101 22:50:27.574451   13644 iscsi_util.go:240] ############### DEBUG: b.Iface: default 
E1101 22:50:27.574490   13644 iscsi_util.go:241] ############### DEBUG: b.Iqn: iqn.2003-01.org.linux-iscsi.minishift:targetd 
E1101 22:50:27.574498   13644 iscsi_util.go:242] ############### DEBUG: b.Portals: [192.168.122.139:3260 :3260] 
E1101 22:50:27.574508   13644 iscsi_util.go:243] ############### DEBUG: b.lun: 0 
E1101 22:50:27.661836   13644 iscsi_util.go:314] iscsi: last error occurred during iscsi init: 
iscsi: failed to sendtargets to portal :3260 output: iscsiadm: Cannot resolve host . getaddrinfo error: [Name or service not known] 
iscsiadm: cannot resolve host name  
iscsiadm: cannot resolve host name  
iscsiadm: No portals found 
, err exit status 21 
E1101 22:50:27.662429   13644 nestedpendingoperations.go:264] Operation for "\"kubernetes.io/iscsi/192.168.122.139:3260:iqn.2003-01.org.linux-iscsi.minishift:targetd:0\"" failed. No retries permitted until 2017-11-01 22:50:28.162384546 -0400 EDT (durationBeforeRetry 500ms). Error: "MapVolume.WaitForAttach failed for volume \"pvc-8c1b77f7-bf78-11e7-a1cf-525400f59db4\" (UniqueName: \"kubernetes.io/iscsi/192.168.122.139:3260:iqn.2003-01.org.linux-iscsi.minishift:targetd:0\") pod \"iscsi-pv-pod1\" (UID: \"92cc6e84-bf78-11e7-a1cf-525400f59db4\") : stat /var/lib/kubelet/plugins/kubernetes.io/iscsi/volumeDevices/iface-default/192.168.122.139:3260-iqn.2003-01.org.linux-iscsi.minishift:targetd-lun-0: no such file or directory" 
E1101 22:50:28.069546   13644 kubelet_volumes.go:128] Orphaned pod "1cced74d-bd2d-11e7-800d-525400f59db4" found, but volume paths are still present on disk : There were a total of 1 errors similar to this. Turn up verbosity to see them. 
```

Portals should be passed like this.

```
@@ -120,6 +121,10 @@ func (p *iscsiProvisioner) Provision(options controller.VolumeOptions) (*v1.Pers
        }
        log.Debugln("volume created with vol and lun: ", vol, lun)
 
+       var portals []string
+       if len(options.Parameters["portals"]) > 0 {
+               portals = strings.Split(options.Parameters["portals"], "'")
+       }
        annotations := make(map[string]string)
        annotations["volume_name"] = vol
        annotations["pool"] = pool
@@ -134,13 +139,14 @@ func (p *iscsiProvisioner) Provision(options controller.VolumeOptions) (*v1.Pers
                Spec: v1.PersistentVolumeSpec{
                        PersistentVolumeReclaimPolicy: options.PersistentVolumeReclaimPolicy,
                        AccessModes:                   options.PVC.Spec.AccessModes,
                        Capacity: v1.ResourceList{
                                v1.ResourceName(v1.ResourceStorage): options.PVC.Spec.Resources.Requests[v1.ResourceName(v1.ResourceStorage)],
                        },
                        PersistentVolumeSource: v1.PersistentVolumeSource{
                                ISCSI: &v1.ISCSIVolumeSource{
                                        TargetPortal:      options.Parameters["targetPortal"],
-                                       Portals:           strings.Split(options.Parameters["portals"], "'"),
+                                       Portals:           portals,
                                        IQN:               options.Parameters["iqn"],
                                        ISCSIInterface:    options.Parameters["iscsiInterface"],
                                        Lun:               lun,
```

/cc @raffaelespazzoli

**Comment by raffaelespazzoli** [comment]
@mtanino thanks. I created a PR to fix the issue as you suggested:
https://github.com/kubernetes-incubator/external-storage/pull/434

while it's getting merge you can find a docker image with the fix here:
`docker pull raffaelespazzoli/iscsi-controller`

**Comment by mtanino** [comment]
closed via #434
